### DISCUSSION TITLE
Jose@3: error TS1109: Expression expected

### DISCUSSION THREAD

**Discussion by cnuss** [panva/jose#118] [Q&A]
Greetings!

This is probably a basic TypeScript config issue but I'm a bit confused on an error after implementing jose@3.1.2.

In short, here's my `src/index.ts` (similar to [this](https://github.com/panva/jose/blob/master/docs/functions/_jwks_remote_.createremotejwkset.md#readme))

```
import jwtVerify from 'jose/jwt/verify';
import createRemoteJWKSet from 'jose/webcrypto/jwks/remote';

const JWKS = createRemoteJWKSet(new URL('https://www.googleapis.com/oauth2/v3/certs'));

const jwt = 'ey...';

jwtVerify(jwt, JWKS).then(result => {
  console.log(result.protectedHeader);
  console.log(result.payload);
});
```

When I run a `tsc`, I get the following error:

```
yarn run v1.22.4
$ tsc
node_modules/jose/src/jwk/parse.ts:60:9 - error TS1109: Expression expected.

60   alg ||= jwk.alg
           ~

node_modules/jose/src/jwk/parse.ts:73:24 - error TS1109: Expression expected.

73       octAsKeyObject ??= jwk.ext !== true
                          ~


Found 2 errors.
```

Here's my tsconfig.json:

```
{
  "compilerOptions": {
    "target": "ES2019",
    "module": "CommonJS",
    "declaration": true,
    "outDir": "./dist",
    "strict": true
  },
  "include": ["src"],
  "exclude": ["node_modules", "**/__tests__/*"]
}
```

Is something misconfigured or is there an issue that I need to report?

**Comment by panva** [comment]
You're importing experimental node webcrypto runtime that only works in node >15.0.0

What lead you to import "jose/webcrypto/jwks/remote"?

**Comment by cnuss** [comment]
Thanks for your quick response @panva ! Good eye! but I've also now tried:

```
import jwtVerify from 'jose/jwt/verify';
import createRemoteJWKSet from 'jose/jwks/remote';

const JWKS = createRemoteJWKSet(new URL('https://www.googleapis.com/oauth2/v3/certs'));

const jwt = 'ey...';

jwtVerify(jwt, JWKS).then(result => {
  console.log(result.protectedHeader);
  console.log(result.payload);
});
```

The same error occurs.

I've also tried node-15.3.0, and have the same error:

```
$ nodeenv -n 15.3.0 ~/node-15.3.0
 * Install prebuilt node (15.3.0) ..... done.
$ source ~/node-15.3.0/bin/activate
$ yarn build
yarn run v1.22.4
$ tsc
node_modules/jose/src/jwk/parse.ts:60:9 - error TS1109: Expression expected.

60   alg ||= jwk.alg
           ~

node_modules/jose/src/jwk/parse.ts:73:24 - error TS1109: Expression expected.

73       octAsKeyObject ??= jwk.ext !== true
                          ~


Found 2 errors.

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Does anything else leap off the page for you? 

I'm also trying to understand what `||=` and `??=` mean in variable assignment...

**Comment by panva** [comment]
The snippet works just fine with regular node. I'm using typescript to generate bundles from these very source files. The syntax is valid and compiled from properly.

Possibly update your typescript version. Else i have no clue and it's simply a tooling issue.

**Comment by cnuss** [comment]
Nailed it @panva thank you!

`yarn add --dev typescript@4` did the trick (specifically I landed on `typescript@4.1.2`)

**Comment by panva** [comment]
Please consider [supporting](https://github.com/panva/jose?sponsor=1) the library if it provides value to you or your company and this support was of help to you. Supporting the library means, amongst other things, that such support will be available in the future.
